### PR TITLE
Support for reverts

### PR DESCRIPTION
**Pull request by 0xZenus**

### CODE REVIEW

File: integration/test/scenarios/core_scenarios/ScenarioA.sol [hunk 1]
```diff
@@ -134,12 +139,39 @@ contract ScenarioCoreA is ScenarioFramework {
...
+            bytes memory inputs = abi.encode();
+
+            try this.take({
+                accountId: account1,
+                user: user1,
+                marketId: marketId,
+                exchangeId: exchangeId,
+                inputs: inputs
+            }) {
+                revert("Unexpected successful transaction");
```
Comment (ioanaroceanu2000) [inline integration/test/scenarios/core_scenarios/ScenarioA.sol:173]: can we do the same thing with vm.expectRevert()? can we specify what the revert error is?

File: integration/test/scenarios/core_scenarios/ScenarioA.sol [hunk 2]
```diff
@@ -16,6 +18,9 @@ import "forge-std/Test.sol";
 contract ScenarioCoreA is ScenarioFramework {
     using SetUtil for SetUtil.Bytes32Set;
 
+    MockERC20 public weth = new MockERC20("weth", "weth", owner, 0);
```
Comment (ioanaroceanu2000) [inline integration/test/scenarios/core_scenarios/ScenarioA.sol:21]: why moved?

**Review by avniculae — APPROVED** [review]

File: integration/test/scenarios/core_scenarios/ScenarioA.sol (hunk 1 above)
Comment (0xZenus) [reply integration/test/scenarios/core_scenarios/ScenarioA.sol]: done, thanks.

File: integration/test/scenarios/core_scenarios/ScenarioA.sol (hunk 2 above)
Comment (0xZenus) [reply integration/test/scenarios/core_scenarios/ScenarioA.sol:21]: caught up on Discord.

File: integration/test/scenarios/core_scenarios/ScenarioA.sol (hunk 1 above)
Comment (0xZenus) [reply integration/test/scenarios/core_scenarios/ScenarioA.sol]: d7db2954fe79ef126cc9852436019d6fb89a967d

**Review by ioanaroceanu2000 — APPROVED** [review]

**Pull request merged**